File: backend/schemas/budget.py
from typing import Optional
from pydantic import BaseModel, Field, ConfigDict


class BudgetCreate(BaseModel):
    amount: float = Field(
        gt=0
    )


class BudgetResponse(BaseModel):
    id: int
    user_id: int
    amount: float

    model_config = ConfigDict(from_attributes=True)


class BudgetStatusResponse(BaseModel):
    budget: float
    estimated_cost: float
    remaining: float
    status: str

File: frontend/src/types/budget.ts
export interface BudgetRecord {
  id: number;
  user_id: number;
  amount: number;
}

export interface BudgetCreate {
  amount: number;
}

export interface BudgetStatusSummary {
  budget: number;
  estimated_cost: number;
  remaining: number;
  status: "OVER_BUDGET" | "NEAR_LIMIT" | "WITHIN_BUDGET" | string;
}
